smt/applications/tests/test_mixed_integer.py

```diff
@@ -107,11 +146,106 @@ def test_cast_to_mixed_integer(self):
             [[-5, 5], ["blue", "red"], ["short", "medium", "long"], [0, 2]],
             dtype="object",
         )
-        x = np.array([1.5, 0, 2, 1])
+        x = np.array([1.5, 0, 2, 1.1])
         self.assertEqual(
             [1.5, "blue", "long", 1], cast_to_mixed_integer(xtypes, xlimits, x)
         )
 
+    def test_unfold_xlimits_with_continuous_limits(self):
+        xtypes = [FLOAT, (ENUM, 2), (ENUM, 3), INT]
+        xlimits = np.array(
+            [[-5, 5], ["blue", "red"], ["short", "medium", "long"], [0, 2]],
+            dtype="object",
+        )
+        l = unfold_xlimits_with_continuous_limits(xtypes, xlimits)
+        self.assertEqual(
+            np.array_equal(
+                [[-5, 5], [0, 1], [0, 1], [0, 1], [0, 1], [0, 1], [0, 2]],
+                unfold_xlimits_with_continuous_limits(xtypes, xlimits),
+            ),
+            True,
+        )
+
+    def test_unfold_xlimits_with_continuous_limits2(self):
```

Comment:
Better to suffix the test name with some information about the specificity of the test, like `test_unfold_xlimits_with_continuous_limits_and_ordinal_values`